src/services/formatting/smartIndenter.ts

```diff
@@ -555,19 +550,33 @@ namespace ts.formatting {
                     return childKind !== SyntaxKind.NamedExports;
                 case SyntaxKind.ImportDeclaration:
                     return childKind !== SyntaxKind.ImportClause ||
-                        ((<ImportClause>child).namedBindings && (<ImportClause>child).namedBindings.kind !== SyntaxKind.NamedImports);
+                        (!!(<ImportClause>child).namedBindings && (<ImportClause>child).namedBindings.kind !== SyntaxKind.NamedImports);
                 case SyntaxKind.JsxElement:
                     return childKind !== SyntaxKind.JsxClosingElement;
             }
             // No explicit rule for given nodes so the result will follow the default value argument
             return indentByDefault;
         }
 
-        /*
-        Function returns true when the parent node should indent the given child by an explicit rule
-        */
-        export function shouldIndentChildNode(parent: TextRangeWithKind, child?: TextRangeWithKind): boolean {
-            return nodeContentIsAlwaysIndented(parent.kind) || nodeWillIndentChild(parent, child, /*indentByDefault*/ false);
+        function isControlFlowEndingStatement(kind: SyntaxKind) {
+            switch (kind) {

```

Comment:
should we also check that it is the last statement of the block?

e.g.:

```ts
switch(x) {
   case 1:
       if (cond)
          break;
          |
}
```

Reply:
We always dedent after a statement in a brace-less `if` already.